libsolidity/analysis/ConstantEvaluator.h

```diff
@@ -48,8 +49,11 @@ class ConstantEvaluator: private ASTConstVisitor
 	virtual void endVisit(BinaryOperation const& _operation);
 	virtual void endVisit(UnaryOperation const& _operation);
 	virtual void endVisit(Literal const& _literal);
+	virtual void endVisit(Identifier const& _identifier);
 
 	ErrorReporter& m_errorReporter;
+	size_t depth; // The recursion depth; to avoid infinite recursion caused by cyclic constant definition
+	size_t DEPTH_LIMIT = 32;
```

Comment:
If this constant is kept here it should be `static const` and probably named `RECURSION_DEPTH_LIMIT`. Though looking at the other PR consensus seems to be including the value in the code instead.